extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js

```diff
@@ -328,6 +328,11 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
       const signals = impl.signals();
       return signals.whenSignal(CommonSignals.INI_LOAD);
     }).then(() => {
+      // Ensures the video-manager does not follow the autoplay attribute on
+      // amp-video tags, which would play the ad in the background before it is
+      // displayed.
+      ampStoryAdPage.getImpl().then(impl => impl.delegateVideoAutoplay());
```

Comment:
Do you think we should also add this in AmpStory's `insertPage` function?  I'm worried we'll forget this issue exists and we'll have a new place in the runtime that inserts a page for some reason that will suffer from the same problems.

Reply:
The issue is due to the media elements being added after the page is created.
In this case, the videos are added even after the `insertPage` method is called...
We could wait for the `amp-ad` to be loaded and rendered (if present) from `insertPage`, but that wouldn't prevent bugs coming from other asynchronous components. :(

Reply:
Oh 😦 

True.